### PR TITLE
DKG migration tool from v1.* to v2.0.0

### PR DESCRIPTION
**Pull request by CluEleSsUK**

### CODE REVIEW

**Comment by CluEleSsUK** [comment]
Merging this because it's develop

**Review by AnomalRoil — COMMENTED** [review]
Wouldn't this belong in https://github.com/drand/drand-tools/ rather?

**Comment by CluEleSsUK** [comment]
It could do, but it may be simpler for the LoE members to all have it in one CLI rather than having to download and run something separate - especially if they _all must_ do it for the v2 transition.
Tbh it's probably worth even automating the migration on startup

File: dkg/store.go [hunk 1]
```diff
@@ -14,8 +18,9 @@ import (
 
 type boltStore struct {
 	sync.RWMutex
-	db  *bolt.DB
-	log log.Logger
+	db            *bolt.DB
+	log           log.Logger
+	migrationLock sync.Mutex
```
Comment (AnomalRoil) [inline dkg/store.go:23]: Do we really need a `migrationLock`? 😢

Comment (CluEleSsUK) [reply dkg/store.go:23]: Eughh yeah unfortunately - access to the `map[string]Broadcast` needs to be protected unfort.
I think when we move to gossip mode we might need to even lock to process each packet so we don't do:
```
current = getCurrentState()
finalState = applyChange(current)
store(finalState)
```

Although I haven't yet looked into bolt transactions for that scenario, they might be a better choice

**Pull request merged**